subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c

```diff
@@ -677,6 +684,51 @@ gst_msdkenc_init_encoder (GstMsdkEnc * thiz)
     gst_msdkenc_add_extra_param (thiz, (mfxExtBuffer *) & ext_vsi);
   }
 
+  if (thiz->string_params) {
+#if (MFX_VERSION >= 2010)
+    ret = MFXVideoCORE_GetHandle (session, MFX_HANDLE_CONFIG_INTERFACE,
+        (mfxHDL *) (&iface));
+    if (ret < 0) {
+      GST_ERROR_OBJECT (session, "Error getting mfx config interface handle");
+      goto failed;
+    }
+
+    string_arr = g_strsplit (thiz->string_params, ",", 0);
+
+    for (guint m = 0; string_arr[m]; m++) {
+      gchar **tmp_arr = g_strsplit (string_arr[m], ":", 0);
+      const gchar *key = tmp_arr[0];
+      const gchar *value = tmp_arr[1];
+      mfxExtBuffer *new_ext_buf;
+
+      for (guint n = 0; n < 2; n++) {
+        ret = iface->SetParameter (iface, (mfxU8 *) key, (mfxU8 *) value,
+            MFX_STRUCTURE_TYPE_VIDEO_PARAM, &thiz->param, &ext_buf);
+        if (ret == MFX_ERR_NONE) {
+          break;
+        } else if (n == 0 && ret == MFX_ERR_MORE_EXTBUFFER) {
+          thiz->extra_string_params =
+              g_realloc_n (thiz->extra_string_params,
+              thiz->num_extra_string_params + 1, sizeof (mfxExtBuffer));
```

Comment:
The size should be sizeof(mfxExtBuffer *) instead of sizeof(mfxExtBuffer)

Reply:
Fixed, thanks